src/main/java/com/sourcegraph/webhook/registry/WebhookRegistry.java

```diff
@@ -78,16 +77,43 @@ public WebhookRegistry(ActiveObjects ao, ProjectService projects, RepositoryServ
     public static void register(Webhook hook) throws WebhookException {
         String scope = resolveScopeID(hook.scope);
 
-        activeObjects.executeInTransaction(() -> {
-            Map<String, Object> params = new HashMap<>();
-            params.put("NAME", hook.name);
-            params.put("SCOPE", scope);
-            params.put("ENDPOINT", hook.endpoint);
-            params.put("SECRET", hook.secret);
+        WebhookEntity[] ents = activeObjects.find(WebhookEntity.class, Query.select().where("ID = ?", hook.id));
+        if (ents.length == 1) {
```

Comment:
Would be more readable to:

```
if (ents.length == 0) {
   create(...)
} else {
   update(...)
}
```

Reply:
Don't we need to wrap the read and write cycle in a transaction? i.e. `activeObjects.find` needs to be ran in the scope of a transaction if its results are going to be modified and written back.